### PR TITLE
ENT-1557 | Updating utils files with helper methods

### PR DESCRIPTION
**Pull request by christopappas**
`get_request_or_stub` (stolen from edx platform) - gets a request
`get_decoded_jwt_from_request` - get a jwt, using request
`request_user_has_implicit_access_via_jwt` - gets access bool, using jwt and role name
`user_has_access_via_database` - gets access bool, using user obj with role name, name of role class, and name of role assignment class
For ENT-1556:
`create_role_auth_claim_for_user` - creates a list of strings containing role & context information for a given user. Roles added are determined by a configuration setting

### CODE REVIEW

File: edx_rbac/utils.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import crum
+
+from edx_rest_framework_extensions.auth.jwt.cookies import (
+    jwt_cookie_name,
```
Comment (brittneyexline) [inline edx_rbac/utils.py]: why isn't this being used below to get the jwt_cookie_name?

File: edx_rbac/utils.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            'enterprise_admin': ['coupon-management', 'data_api_access'],
+            'enterprise_leaner': [],
+            'portal_admin_no_coupons': ['data_api_access']
+        }
+    """
+    jwt_roles_claim = decoded_jwt.get('roles', [])
+
+    feature_roles = []
+    for role_data in jwt_roles_claim:
+        role_in_jwt, object_type, object_key = role_data.split(':') # split should be more robust because of our cousekeys having colons
```
Comment (brittneyexline) [inline edx_rbac/utils.py]: i haven't implemented it yet, but i think we'll probably get rid of the object_type and just have role:context_data

File: edx_rbac/utils.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+def user_has_access_via_database(user, role_name, role_class, role_assignment_class):
+    """
+    Checks if there is a role assignment for a given user and role.
+
+    The role object itself is found via the role_name
+    """
+
+    role_obj = role_class.objects.get(name=role_name)
+    try:
+        role_assignment_class.objects.get(user=user, role=role_obj)
```
Comment (brittneyexline) [inline edx_rbac/utils.py]: you can get rid of the first line and just do: `role_assignment_class.objects.get(user=user, role__name=role_name)`

File: edx_rbac/utils.py (hunk 1 above)
Comment (christopappas) [reply edx_rbac/utils.py]: ah, probably a copy-paste thing when i was extracting the code from the protoytype. this function is actually just 1 line... i wonder if it's even worth importing

File: edx_rbac/utils.py (hunk 2 above)
Comment (christopappas) [reply edx_rbac/utils.py]: cool, noted!

File: edx_rbac/utils.py (hunk 3 above)
Comment (christopappas) [reply edx_rbac/utils.py]: pushing up in next commit

File: edx_rbac/utils.py (hunk 1 above)
Comment (christopappas) [reply edx_rbac/utils.py]: i did end up making this change, so resolving!

File: edx_rbac/utils.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    for system_role_class in settings.SYSTEM_WIDE_ROLE_CLASSES:
+        app_name, model_name = system_role_class.split('.')
+        model_class = apps.get_model(app_name, model_name)
+
+        role_assignments = model_class.objects.filter(
+            user=user
+        ).select_related('role')
+
+        for role_assignment in role_assignments:
+            role_string = '{role}:{context}'.format(
```
Comment (brittneyexline) [inline edx_rbac/utils.py]: i think we should add some logic here to check if the context is set to something, because not all roles will map to a context

Comment (christopappas) [reply edx_rbac/utils.py]: ah right. in the case of no context, just a role, and no colon?

Comment (brittneyexline) [reply edx_rbac/utils.py]: yep

**Pull request merged**